src/mimeparser.rs

```diff
@@ -322,6 +336,29 @@ impl<'a> MimeParser<'a> {
         Ok(())
     }
 
+    fn avatar_action_from_header(&mut self, header_value: String) -> AvatarAction {
+        if header_value == "0" {
+            return AvatarAction::Delete;
+        } else {
+            let mut i = 0;
+            while i != self.parts.len() {
+                let part = &mut self.parts[i];
+                if let Some(part_filename) = &part.org_filename {
+                    if part_filename == &header_value {
+                        if let Some(blob) = part.param.get(Param::File) {
+                            let res = AvatarAction::Change(blob.to_string());
+                            self.parts.remove(i);
+                            return res;
+                        }
+                        break;
+                    }
+                }
+                i += 1;
```

Comment:
hum, isn't there a chance of a run-away or index-error panic here? (note that accessing out-of-bound indexes results in panic, you can use "Vec::get() i think to get an Option". 
 
With  ` i == 4 &&self.parts.len() ==5` , the loop-body will excecute. If it removes the part, then next while-check would be `i==5 && self.parts.len() == 4`, and then index-out-of-bound-panic.

Reply:
ah, nevermind. i overlooked the "return res;".  phew :)

Reply:
still given a generally defensive "unwrap_or_default()" coding style,  `[i]` indexing is dangerous :)

Reply:
i agree, that it would be nice to drop the index-counter and use for foreach or so. but the index is needed for remove().
there is this nightly feature that may help, https://doc.rust-lang.org/std/vec/struct.Vec.html#method.drain_filter , however, it is marked as experimental - and this code was already used for ages, so maybe not change it to "experimental"